src/main/java/com/mindsdb/kafka/connect/client/MindsDBCloudClientImpl.java

```diff
@@ -0,0 +1,109 @@
+package com.mindsdb.kafka.connect.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.mindsdb.kafka.connect.client.models.Predictor;
+import org.apache.kafka.connect.errors.ConnectException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class MindsDBCloudClientImpl implements MindsDBClient {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    public static final String CONNECTION_ERROR = "Failed to send request to MindsDB server";
+
+    private final HttpClient httpClient;
+    private final MindsDBClientConfig clientConfig;
+
+    public MindsDBCloudClientImpl(HttpClient httpClient, MindsDBClientConfig clientConfig) {
+        this.httpClient = httpClient;
+        this.clientConfig = clientConfig;
+
+        try {
+            postToMindsDb(
```

Comment:
I think that `postToMindsDb` need to return response headers or cookies because we need to get `session` cookie from `cloud/login/` response.

Reply:
I think that having a separate method for `login` is useful, cause we need to call this method again when `session` cookie will expire. And ceiling `postToMindsDb` with same set of parameters in several places won't look good.